From 0b1203f39d3138d13216cd5464c03e321ad67250 Mon Sep 17 00:00:00 2001
From: Xu Qiang <xuqiang36@huawei.com>
Date: Mon, 13 Feb 2023 17:18:10 +0800
Subject: [PATCH 014/500] [Huawei] mm/sharepool: Add meminfo_k2u_size.

Offering: HULK
hulk inclusion
category: other
bugzilla: https://gitee.com/openeuler/kernel/issues/I6ET9W

----------------------------------------------

Add meminfo_k2u_size and delete duplicate or similar code.

Signed-off-by: Xu Qiang <xuqiang36@huawei.com>
---
 mm/share_pool.c | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 73ebdf265e5f..52af832c8fd5 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -300,6 +300,11 @@ static inline long meminfo_alloc_sum_byKB(struct sp_meminfo *meminfo)
 	return byte2kb(meminfo_alloc_sum(meminfo));
 }
 
+static inline long meminfo_k2u_size(struct sp_meminfo *meminfo)
+{
+	return byte2kb(atomic64_read(&meminfo->k2u_size));
+}
+
 static unsigned long sp_mapping_type(struct sp_mapping *spm)
 {
 	return spm->type;
@@ -3618,11 +3623,6 @@ static void get_mm_rss_info(struct mm_struct *mm, unsigned long *anon,
 	*total_rss = *anon + *file + *shmem;
 }
 
-static long get_proc_k2u(struct sp_meminfo *meminfo)
-{
-	return byte2kb(atomic64_read(&meminfo->k2u_size));
-}
-
 static void get_process_sp_res(struct sp_group_master *master,
 		long *sp_res_out, long *sp_res_nsize_out)
 {
@@ -3657,11 +3657,6 @@ static void get_process_non_sp_res(unsigned long total_rss, unsigned long shmem,
 	*non_sp_shm_out = non_sp_shm;
 }
 
-static long get_spg_proc_k2u(struct sp_group_node *spg_node)
-{
-	return byte2kb(atomic64_read(&spg_node->meminfo.k2u_size));
-}
-
 static void print_process_prot(struct seq_file *seq, unsigned long prot)
 {
 	if (prot == PROT_READ)
@@ -3708,7 +3703,7 @@ int proc_sp_group_state(struct seq_file *m, struct pid_namespace *ns,
 	seq_printf(m, "%-8d %-16s %-9ld %-9ld %-9ld %-10ld %-10ld %-8ld\n",
 		   master->tgid, master->comm,
 		   meminfo_alloc_sum_byKB(meminfo),
-		   get_proc_k2u(meminfo),
+		   meminfo_k2u_size(meminfo),
 		   sp_res, non_sp_res, non_sp_shm,
 		   page2kb(mm->total_vm));
 
@@ -3720,7 +3715,7 @@ int proc_sp_group_state(struct seq_file *m, struct pid_namespace *ns,
 		seq_printf(m, "%-8d %-9ld %-9ld %-9ld ",
 				spg_node->spg->id,
 				meminfo_alloc_sum_byKB(&spg_node->meminfo),
-				get_spg_proc_k2u(spg_node),
+				meminfo_k2u_size(&spg_node->meminfo),
 				meminfo_alloc_sum_byKB(&spg_node->spg->meminfo));
 		print_process_prot(m, spg_node->prot);
 		seq_putc(m, '\n');
@@ -3928,7 +3923,7 @@ static int proc_usage_by_group(int id, void *p, void *data)
 		seq_printf(seq, "%-8d ", id);
 		seq_printf(seq, "%-9ld %-9ld %-9ld %-8ld %-7ld %-7ld ",
 				meminfo_alloc_sum_byKB(&spg_node->meminfo),
-				get_spg_proc_k2u(spg_node),
+				meminfo_k2u_size(&spg_node->meminfo),
 				meminfo_alloc_sum_byKB(&spg_node->spg->meminfo),
 				page2kb(mm->total_vm), page2kb(total_rss),
 				page2kb(shmem));
@@ -3991,7 +3986,7 @@ static int proc_usage_show(struct seq_file *seq, void *offset)
 		seq_printf(seq, "%-8d %-16s %-9ld %-9ld %-9ld %-10ld %-10ld %-8ld\n",
 				master->tgid, master->comm,
 				meminfo_alloc_sum_byKB(meminfo),
-				get_proc_k2u(meminfo),
+				meminfo_k2u_size(meminfo),
 				sp_res, non_sp_res, non_sp_shm,
 				page2kb(master->mm->total_vm));
 	}
-- 
2.17.1

